spec/lib/rabbit_feed/event_spec.rb

```diff
@@ -67,6 +67,57 @@ module RabbitFeed
       end
     end
 
+    describe '#serialize' do
+      let(:schema) do
+        Avro::Schema.parse(
+          {
+            name:   'example-1.0',
+            type:   'record',
+            fields: [
+              {
+                name: 'payload',
+                type: {
+                  name: 'event_payload',
+                  type: 'record',
+                  fields: [{ name: 'event_field', type: 'int' }],
+                },
+              },
+              {
+                name: 'metadata',
+                type: {
+                  name: 'event_metadata',
+                  type: 'record',
+                  fields: [{ name: 'name', type: 'string' }],
+                },
+              },
+            ],
+          }.to_json
+        )
+      end
+
+      context 'with invalid payload' do
+        let(:payload)  { { 'event_field' => 'INCORRECT' } }
```

Comment:
> dare I say it - stub something to force it to throw an exception?

No you may not suggest that! I want to avoid stubbing. I will use multiple fields though to better demonstrate/test